hermes/src/api.rs

```diff
@@ -103,19 +108,22 @@ pub async fn run(opts: RunOptions, state: Arc<State>, mut update_rx: Receiver<()
         // default value for this parameter).
         .layer(Extension(QsQueryConfig::new(5, false)));
 
-    // Call dispatch updates to websocket every 1 seconds
-    // FIXME use a channel to get updates from the store
     tokio::spawn(async move {
         while !crate::SHOULD_EXIT.load(Ordering::Acquire) {
-            // Causes a full application shutdown if an error occurs, we can't recover from this so
-            // we just quit.
-            if update_rx.recv().await.is_none() {
-                tracing::error!("Failed to receive update from store.");
-                crate::SHOULD_EXIT.store(true, Ordering::Release);
-                break;
+            let event = update_rx.recv().await;
+            match event {
```

Comment:
Nitpick but its short enough to just:

```
match update_rx.recv().await {
```

Less variables and `event` below no longer shadows.